packages/vscode-extension/src/builders/utils.ts

```diff
@@ -0,0 +1,52 @@
+import { readdir } from "fs/promises";
+import path from "path";
+import { Logger } from "../Logger";
+import { exec } from "../utilities/subprocess";
+import { CancelToken } from "./cancelToken";
+import { DevicePlatform } from "../common/DeviceManager";
+
+export function isAppFile(name: string) {
+  return name.endsWith(".app");
+}
+
+export function isApkFile(name: string) {
+  return name.endsWith(".apk");
+}
+
+export async function extractTarApp(
+  binaryPath: string,
+  pathToExtract: string,
+  cancelToken: CancelToken,
+  platform: DevicePlatform
+) {
+  const { failed } = await cancelToken.adapt(
+    tarCommand({ archivePath: binaryPath, extractDir: pathToExtract })
+  );
+
+  if (failed) {
+    Logger.error(`Failed to extract archive '${binaryPath}' to '${pathToExtract}'.`);
+    return undefined;
+  }
+
+  // assuming that the archive contains only one app file
+  const appName = (await readdir(pathToExtract)).find(
```

Comment:
instead of having this here both here and in runExternalBuild maybe we should just make this metod return a single extracted file and then only make the file extension checks in `runExternalBuild`